Functions/Public/Get-GitHubOrganization.ps1

```diff
@@ -1,28 +1,87 @@
 function Get-GitHubOrganization {
-    [OutputType('PSGitHub.Organization')]
-    [CmdletBinding()]
-    param (
-        # Gets the org a specific user is part of.
-        [Parameter(ValueFromPipelineByPropertyName, ParameterSetName = 'User')]
-        [string] $Username,
-
-        # Gets the org with a specific name.
-        [Parameter(ValueFromPipelineByPropertyName, ParameterSetName = 'Org')]
-        [string] $OrganizationName,
-
-        [Security.SecureString] $Token
-    )
-
-    $url = if ($Username) {
-        "/users/$Username/orgs"
-    } elseif ($OrganizationName) {
-        "/orgs/$OrganizationName"
-    } else {
-        "/organizations"
-    }
-
-    Invoke-GitHubApi $url -Token $Token | ForEach-Object { $_ } | ForEach-Object {
-        $_.PSTypeNames.Insert(0, 'PSGitHub.Organization')
-        $_
-    }
+	<#
+	.SYNOPSIS
+	Retrieve a GitHub Organization.
+
+	.PARAMETER UserName
+	The GitHub username/handle to query for organization memberships
+
+	.PARAMETER Name
+	The GitHub organization name to query against
+
+	.PARAMETER Token
+	The secure string representing a token used for authentication against the API
+	#>
+	[OutputType('PSGitHub.Organization')]
+	[CmdletBinding()]
+	param (
+		# Gets the org a specific user is part of.
+		[Parameter(ParameterSetName = 'UserMemberships',
+			Mandatory = $false, 
+			ValueFromPipelineByPropertyName = $true)]
+		[Alias('Handle')]
+		[string]$UserName,
+
+	   # Hostname of the GitHub Enterprise Server ("GHES")
+	   [Parameter(Mandatory = $false, 
+		   ValueFromPipelineByPropertyName = $false)]
+	   [Alias('ComputerName', 'Host')]
+		[String]$HostName,
+
+		# Gets the org with a specific name.
+		[Parameter(ParameterSetName = 'Org',
+			Mandatory = $false,
+			ValueFromPipelineByPropertyName = $true)]
+		[Alias('OrganizationName')]
+		[string[]]$Name,
+
+		# Secure token to use in authentication for this operation
+		[Security.SecureString]$Token
+	)
+
+	Begin {
+
+		Write-Debug -Message "Entered Function: Get-GitHubOrganization"
+
+		If ($HostName) {
+			Write-Debug -Message "Hostname is: $HostName"
+			$BaseUrl = "https://$HostName/api/v3"
+			Write-Debug -Message "BaseUrl is: $BaseUrl"
+		}
+
+		Else {
+			Write-Debug -Message 'Setting BaseUrl to GitHub.com'
+			$BaseUrl = 'https://api.github.com'
```

Comment:
This logic should be implemented in `Invoke-GitHubApi` since it applies to all API requests, no t just orgs

Reply:
Sorry for the confusion here - my understanding is that the logic is in the `Invoke-GitHubApi` on line 58

`$Uri = [Uri]::new([Uri]::new('https://api.github.com'), $Uri)`

Basically if `$Uri` is a path, then it'll append that to `https://api.github.com/my/path` but if its sent a full URL including a different FQDN, then it'll remove `api.github.com` hostname and replace it with the one specified in the property value.

This enables us to have the calling Cmdlet be flexible and make the determination. I'm happy to jump on a call to discuss this though.

Reply:
Yes, but instead of any `if`/`else` here, we could just add the `HostName` parameter to `Invoke-GitHubApi` and pass the `HostName` through (same as with `Token`).

Reply:
I see, something like this -

**Invoke-GitHubApi**
```PowerShell
[Parameter(Mandatory = $false)]
[String]$HostName

<...code...>

    If ($HostName) {
        Write-Debug -Message "Hostname is: $HostName"
        $Uri = [Uri]::new([Uri]::new("https://$HostName"), "/api/v3$Uri")
    }
    Else {
        Write-Debug -Message 'Setting BaseUrl to GitHub.com'
        $Uri = [Uri]::new([Uri]::new('https://api.github.com'), $Uri)
    }
```